src/L2/LaunchAuctionPriceOracle.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {FixedPointMathLib} from "solady/utils/FixedPointMathLib.sol";
+
+import {EDAPrice} from "src/lib/EDAPrice.sol";
+import {StablePriceOracle} from "src/L2/StablePriceOracle.sol";
+
+/// @title Launch Auction Price Oracle
+///
+/// @notice The mechanism by which names are auctioned upon Basenames public launch. The RegistrarController
+///     Passes the `launchTime` in place of expiry for all new names. The half life of this contract is hard-coded
+///     to 1.5 hours, accomplished by bitshifting the `endValue` and by passing this period into the exponential decay
+///     calculation.
+///
+///     Inspired by the `ExponentialPremiumPriceOracle` implemented by ENS:
+///     https://github.com/ensdomains/ens-contracts/blob/staging/contracts/ethregistrar/ExponentialPremiumPriceOracle.sol
+///
+/// @author Coinbase (https://github.com/base-org/usernames)
+contract LaunchAuctionPriceOracle is StablePriceOracle {
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                          STORAGE                           */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+    /// @notice Starting premium for the dutch auction.
+    uint256 public immutable startPremium;
+
+    /// @notice Ending value of the auction, calculated on construction.
+    uint256 public immutable endValue;
```

Comment:
Immutable variables are not actually part of contract's storage if that's the intention of `STORAGE` comment.